README.md

```diff
@@ -121,15 +121,75 @@ but it doesn't traverse the object recursively converting nested items:
 ```
 
 
-## Public API
+## API
+
+### Extending `itemadapter`
```

Comment:
I think this section makes more sense at the end of the API section, most people won’t be looking for this.

It may also make sense to make it a section of its own, instead of a part of the API, even if it has its own API. But I’m less sure about this suggestion.